modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationParameters.java

```diff
@@ -0,0 +1,244 @@
+import com.opengamma.strata.calc.config.Measure;
+import com.opengamma.strata.collect.MapStream;
+
+/**
+ * The calculation parameters.
+ * <p>
+ * This provides a set of parameters that will be used in a calculation.
+ * Each parameter defines a {@linkplain CalculationParameter#queryType() query type},
+ * thus the functions are keyed in a {@code Map} by the query type {@code Class}.
+ */
+@BeanDefinition(style = "light")
+public final class CalculationParameters implements ImmutableBean, Serializable {
+
+  /**
+   * An empty instance.
+   */
+  private static final CalculationParameters EMPTY = new CalculationParameters(ImmutableMap.of());
+
+  /**
+   * The parameters, keyed by query type.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ImmutableMap<Class<?>, CalculationParameter> parameters;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an empty instance with no parameters.
+   * 
+   * @return the empty instance
+   */
+  public static CalculationParameters empty() {
+    return EMPTY;
+  }
+
+  /**
+   * Obtains an instance from the specified parameters.
+   * <p>
+   * The list will be converted to a {@code Map} using {@link CalculationParameter#queryType()}.
+   * Each parameter must refer to a different query type.
+   * 
+   * @param parameters  the parameters
+   * @return the calculation parameters
+   */
+  public static CalculationParameters of(CalculationParameter... parameters) {
+    if (parameters.length == 0) {
+      return EMPTY;
+    }
+    return new CalculationParameters(Stream.of(parameters).collect(toImmutableMap(p -> p.queryType())));
+  }
+
+  /**
+   * Obtains an instance from the specified parameters.
+   * <p>
+   * The list will be converted to a {@code Map} using {@link CalculationParameter#queryType()}.
+   * Each parameter must refer to a different query type.
+   * 
+   * @param parameters  the parameters
+   * @return the calculation parameters
+   */
+  public static CalculationParameters of(List<? extends CalculationParameter> parameters) {
+    if (parameters.isEmpty()) {
+      return EMPTY;
+    }
+    return new CalculationParameters(parameters.stream().collect(toImmutableMap(p -> p.queryType())));
+  }
+
+  // create checking for empty
+  private CalculationParameters of(Map<Class<?>, CalculationParameter> map) {
+    if (map.isEmpty()) {
+      return EMPTY;
+    }
+    return new CalculationParameters(map);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Combines this set of parameters with the specified set.
+   * <p>
+   * This set of parameters takes priority.
+   * 
+   * @param other  the other parameters
+   * @return the combined calculation parameters
+   */
+  public CalculationParameters combinedWith(CalculationParameters other) {
+    Map<Class<?>, CalculationParameter> map = new HashMap<>(other.getParameters());
```

Comment:
Return `other` if these parameters are empty? That will happen a lot because this method is used to combine a column's parameters with the default parameters.